acceptance/tests/segments/segments_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package segments
+
+import (
+	"github.com/hashicorp/consul-k8s/acceptance/framework/connhelper"
+	"testing"
+
+	"github.com/hashicorp/consul-k8s/acceptance/framework/consul"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/helpers"
+)
+
+// TestSegments_MeshWithAgentfulClients is a simple test that verifies that
+// the Consul service mesh can be configured to use segments with:
+// - one cluster with an alpha segment configured on the servers
+// - clients enabled and joining the alpha segment
+// - static client can communicate with static server
+func TestSegments_MeshWithAgentfulClients(t *testing.T) {
+	cases := map[string]struct {
+		secure bool
+	}{
+		"not-secure": {secure: false},
+		"secure":     {secure: true},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			cfg := suite.Config()
+			if !cfg.EnableEnterprise {
+				t.Skipf("skipping this test because -enable-enterprise is not set")
+			}
+			ctx := suite.Environment().DefaultContext(t)
+
+			releaseName := helpers.RandomName()
+
+			helmValues := map[string]string{
+				"connectInject.enabled": "true",
+
+				"server.replicas":    "3",
+				"server.extraConfig": `"{\"segments\": [{\"name\":\"alpha1\"\,\"bind\":\"0.0.0.0\"\,\"port\":8303}]}"`,
+
+				"client.enabled":     "true",
+				"client.join[0]":     "${CONSUL_FULLNAME}-server-0.${CONSUL_FULLNAME}-server.${NAMESPACE}.svc:8303",
+				"client.join[1]":     "${CONSUL_FULLNAME}-server-1.${CONSUL_FULLNAME}-server.${NAMESPACE}.svc:8303",
+				"client.join[2]":     "${CONSUL_FULLNAME}-server-2.${CONSUL_FULLNAME}-server.${NAMESPACE}.svc:8303",
```

Comment:
this is needed in whichever consul-k8s versions we backport to (even < 1.0) because of this [this block](https://github.com/hashicorp/consul-k8s/blob/main/charts/consul/templates/client-daemonset.yaml#L378-L388) in client-daemonset.yaml that unless you set this, there is not a way to pass the port you configured for the segment to client.

Reply:
i can add a comment in the test.

Reply:
are you asking if segments will work on agentless?  If so, I am not familiar enough with agentless to know whether this can work.

Reply:
(comment added in a08365770272cd529a9658df9ec1830a04476d38)